client/src/main/java/com/scalar/dl/client/service/GenericContractClientService.java

```diff
@@ -434,6 +444,301 @@ public LedgerValidationResult validateLedger(
     return clientService.validateLedger(buildAssetId(type, keys), startAge, endAge);
   }
 
+  /**
+   * Validates the specified object in the ledger.
+   *
+   * @param objectId an object ID
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateObject(String objectId) {
+    return clientService.validateLedger(buildAssetId(AssetType.OBJECT, ImmutableList.of(objectId)));
+  }
+
+  /**
+   * Validates the specified object in the ledger.
+   *
+   * @param objectId an object ID
+   * @param startAge an age to be validated from (inclusive)
+   * @param endAge an age to be validated to (inclusive)
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateObject(String objectId, int startAge, int endAge) {
+    return clientService.validateLedger(
+        buildAssetId(AssetType.OBJECT, ImmutableList.of(objectId)), startAge, endAge);
+  }
+
+  /**
+   * Validates the specified collection in the ledger.
+   *
+   * @param collectionId a collection ID
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateCollection(String collectionId) {
+    return clientService.validateLedger(
+        buildAssetId(AssetType.COLLECTION, ImmutableList.of(collectionId)));
+  }
+
+  /**
+   * Validates the specified collection in the ledger.
+   *
+   * @param collectionId a collection ID
+   * @param startAge an age to be validated from (inclusive)
+   * @param endAge an age to be validated to (inclusive)
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateCollection(String collectionId, int startAge, int endAge) {
+    return clientService.validateLedger(
+        buildAssetId(AssetType.COLLECTION, ImmutableList.of(collectionId)), startAge, endAge);
+  }
+
+  /**
+   * Validates the schema of the specified table in the ledger.
+   *
+   * @param tableName a table name
+   * @return {@link LedgerValidationResult}
+   * @throws ClientException if a request fails for some reason
+   */
+  public LedgerValidationResult validateTableSchema(String tableName) {
```

Comment:
It might be less error-prone to separate the validation methods for table-oriented and the other.
Let's discuss it.

Reply:
@feeblefakie Thank you for the suggestion! As discussed, let's keep it as is for now and continue discussing introducing a dedicated client service like `TableOrientedClientService`.